templates/intel_pmu.conf.j2

```diff
@@ -10,8 +10,9 @@ LoadPlugin intel_pmu
 {% if collectd_plugin_intelpmu_cores is defined %}
   Cores{% for item in collectd_plugin_intelpmu_cores %} "{{ item }}"{% endfor %}
 {% endif %}
-
+{% if collectd_plugin_intelpmu_reporthardwarecacheevents is defined %}
   ReportHardwareCacheEvents {{ collectd_plugin_intelpmu_reporthardwarecacheevents | string | lower }}
+{% endif %}
   ReportKernelPMUEvents {{ collectd_plugin_intelpmu_reportkernelpmuevents | string | lower }}
```

Comment:
I think you want to wrap this in the same `is defined` no?  https://github.com/voxpupuli/puppet-collectd/blob/master/templates/plugin/intel_pmu.conf.erb#L5-L7